timesketch/api/v1/resources/export.py

```diff
@@ -0,0 +1,175 @@
+# Copyright 2025 Google Inc. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Export resources for version 1 of the Timesketch API."""
+
+import json
+import logging
+
+from flask import request
+from flask import abort
+from flask import Response
+from flask import stream_with_context
+from flask import current_app
+from flask_restful import Resource
+from flask_login import login_required
+from flask_login import current_user
+
+from timesketch.api.v1 import resources
+from timesketch.lib import utils
+from timesketch.lib import forms
+from timesketch.lib.definitions import HTTP_STATUS_CODE_NOT_FOUND
+from timesketch.lib.definitions import HTTP_STATUS_CODE_FORBIDDEN
+from timesketch.lib.definitions import HTTP_STATUS_CODE_BAD_REQUEST
+from timesketch.lib.definitions import DEFAULT_SOURCE_FIELDS
+from timesketch.lib.datastores.opensearch import OpenSearchDataStore
+from timesketch.models.sketch import Sketch
+
+
+logger = logging.getLogger("timesketch.api.export")
+
+DEFAULT_POOL_MAXSIZE = 60
+
+
+class ExportListResource(resources.ResourceMixin, Resource):
+    """Resource to export all events for a sketch."""
+
+    @property
+    def datastore(self):
+        """Property to get an instance of the datastore backend.
+
+        We override the default datastore property to ensure we have a
+        connection pool large enough to support the multiple threads
+        used during sliced export.
+
+        Returns:
+            Instance of lib.datastores.opensearch.OpenSearchDatastore
+        """
+        # Default to 60 connections to align with the LLM Log Analyzer implementation
+        pool_maxsize = current_app.config.get(
+            "OPENSEARCH_SLICED_EXPORT_POOL_MAXSIZE", DEFAULT_POOL_MAXSIZE
+        )
+        return OpenSearchDataStore(pool_maxsize=pool_maxsize)
+
+    @login_required
+    def post(self, sketch_id):
```

Comment:
From the PR description it sounds like this method should be handled with care and is for a specific use case, maybe worth to explain a bit more in the docstring what the purpose, advantage and risk of the method is? And why not use the already existing explore function?